_P251_PZEM004Tv3.ino

```diff
@@ -143,8 +144,9 @@ boolean Plugin_251(byte function, struct EventStruct *event, String& string)
         }
       }
       else
-      {
-        addHtml(F("Tx Pin and Rx Pin have no effect on the configuration as this PZEM is not the main configured."));
+      {        
+        Device[deviceCount].Type = DEVICE_TYPE_DUMMY;   //Erase GPIO choice if not first PZEM

```

Comment:
You know the device type is also used to determine the number of output values?
And the `Device` vector contains the plugin definitions, not the task definitions.
This will have an effect on all instances of the same plugin on an ESP node.

Reply:
For me, "type" only defines the number of datapin (Vtype for # of output):
#define DEVICE_TYPE_SINGLE                  1  // connected through 1 datapin
#define DEVICE_TYPE_DUAL                    2  // connected through 2 datapins
#define DEVICE_TYPE_TRIPLE                  3  // connected through 3 datapins
Am I wrong ?

Thus how can I remove the data pin selection on other instance of the same pluggin (any example on an existing pluggin)?

Reply:
Hmm, you're right it is about the pins, not the values.
But still, values in that vector are for all instances of the same plugin, not per task.
If you need to spilt it, you can set it to some "no pins" value, but then you must add the pin configuration yourself in the PLUGIN_LOAD (and PLUGIN_SAVE) based on other settings of the task config.
Or you can make 2 versions of the plugin, which do share a common C++ object to communicate with the hardware.
I think you should have a single central object shared among several task instances when communicating over a shared bus.
This is also an issue for the Dallas sensors.

The Eastron plugin does just allow multiple instances as long as you set all to use the same pins. There is a shared object which does the actual communication.
But there you may want multiple task instances to collect different values, since those output way more values than  the 4 we can handle on a single task.